include/index.h

```diff
@@ -79,7 +79,7 @@ template <typename T, typename TagT = uint32_t, typename LabelT = uint32_t> clas
 
     // Constructor for incremental index
     DISKANN_DLLEXPORT Index(Metric m, const size_t dim, const size_t max_points, const bool dynamic_index,
-                            const Parameters &indexParameters, const Parameters &searchParameters,
+                            const IndexWriteParameters &indexParameters, const IndexReadParameters &searchParameters,
```

Comment:
Shall we call these classes IndexBuildParameters and IndexSearchParameters?
Also, do we really need IndexSearchParameters in the ctor? Can we not just pass the number of threads to create scratch space for? The Ls parameter is passed directly in search anyway.

Reply:
I would prefer to pass fewer of these params objects around (if you notice I did make an executive decision to do that on one of the method/function calls, `prune_all_neighbors` I think.)  I'd much prefer not to have extra objects if we don't need them

Reply:
> Shall we call these classes IndexBuildParameters ...

I went with IndexWriteParameters because it's for both the major build stuff and any incremental/dynamic changes too.  I didn't want to imply it was only used for building the whole index, and IndexMutationParameters was too dang long.  I don't really have strong feelings about this though

Reply:
I think of index building as both a batch and incremental operation. Your call on this.

Also, can we rename indexParameters to write_params or build_params ?

Reply:
yeah, I'll rename it in a future PR - I like write_params